src/main/java/com/ibanity/apis/client/http/service/impl/IbanityHttpSignatureServiceImpl.java

```diff
@@ -44,34 +40,110 @@ public class IbanityHttpSignatureServiceImpl implements IbanityHttpSignatureServ
     private static final String ACCEPTED_HEADERS_REGEX = "(authorization|ibanity.*?)";
     private static final Pattern HEADERS_PATTERN = Pattern.compile(ACCEPTED_HEADERS_REGEX, Pattern.CASE_INSENSITIVE);
     private static final String SIGNATURE_HEADER_ALGORITHM = "hs2019";
-
     private Clock clock;
     private String certificateId;
     private PrivateKey privateKey;
-    private X509Certificate certificate;
     private String ibanityEndpoint;
+    private String proxyEndpoint;
 
+    /**
+     *
+     * @deprecated certificate parameter is not used anymore.
+     */
     public IbanityHttpSignatureServiceImpl(
             @NonNull PrivateKey privateKey,
             @NonNull X509Certificate certificate,
             @NonNull String certificateId,
             @NonNull String ibanityEndpoint) {
-        this(privateKey, certificate, certificateId, Clock.systemUTC(), ibanityEndpoint);
+        this(privateKey, certificate, certificateId, Clock.systemUTC(), ibanityEndpoint, null);
+    }
+
+    public IbanityHttpSignatureServiceImpl(
+            @NonNull PrivateKey privateKey,
+            @NonNull String certificateId,
+            @NonNull String ibanityEndpoint) {
+        this(privateKey, certificateId, Clock.systemUTC(), ibanityEndpoint, null);
     }
 
+    /**
+     *
+     * @deprecated certificate parameter is not used anymore.
+     */
+    public IbanityHttpSignatureServiceImpl(
+            @NonNull PrivateKey privateKey,
+            @NonNull X509Certificate certificate,
+            @NonNull String certificateId,
+            @NonNull String ibanityEndpoint,
+            @NonNull String proxyEndpoint) {
+        this(privateKey, certificate, certificateId, Clock.systemUTC(), ibanityEndpoint, proxyEndpoint);
+    }
+
+    public IbanityHttpSignatureServiceImpl(
+            @NonNull PrivateKey privateKey,
+            @NonNull String certificateId,
+            @NonNull String ibanityEndpoint,
+            @NonNull String proxyEndpoint) {
+        this(privateKey, certificateId, Clock.systemUTC(), ibanityEndpoint, proxyEndpoint);
+    }
+
+    /**
+     *
+     * @deprecated certificate parameter is not used anymore.
+     */
     public IbanityHttpSignatureServiceImpl(
             @NonNull PrivateKey privateKey,
             @NonNull X509Certificate certificate,
             @NonNull String certificateId,
             @NonNull Clock clock,
             @NonNull String ibanityEndpoint) {
         this.privateKey = privateKey;
-        this.certificate = certificate;
         this.certificateId = certificateId;
         this.clock = clock;
         this.ibanityEndpoint = ibanityEndpoint;
     }
 
+    public IbanityHttpSignatureServiceImpl(
+            @NonNull PrivateKey privateKey,
+            @NonNull String certificateId,
+            @NonNull Clock clock,
+            @NonNull String ibanityEndpoint) {
+        this.privateKey = privateKey;
+        this.certificateId = certificateId;
```

Comment:
should be replaced by `this(privateKey, certificateId, Clock.systemUTC(), ibanityEndpoint, null);`

Reply:
fixed